test/unit/puppetlabs/services/master/master_core_test.clj

```diff
@@ -80,7 +82,100 @@
                     "non-alphanumeric characters")
         (logging/with-test-logging
          (is (= 400 (:status (request
-                              "/v3/environment_classes?environment=~")))))))))
+                              "/v3/environment_classes?environment=~"))))))
+      (testing "calculates etag properly for response payload"
```

Comment:
I found the serialization order / hash equivalence tests to be easier to do at this level than at the full-stack integration level.